subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -115,8 +116,11 @@ static void handle_wifi_scan_result(struct net_mgmt_event_callback *cb)
 		   "Num", "SSID", "(len)", "Chan (Band)", "RSSI", "Security", "BSSID", "MFP");
 	}
 
+	strncpy(ssid_print, entry->ssid, entry->ssid_length);
```

Comment:
We should not trust the supplied ssid_lengh as it could contain garbage. Better to use the max length of the target buffer
```
strncpy(ssid_print, entry->ssid, sizeof(ssid_print) - 1);
ssid_print[sizeof(ssid_print)] = '\0';
```